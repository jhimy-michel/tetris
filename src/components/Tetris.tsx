import React, { useState } from 'react';

import { createStage } from '../gameHelpers';
//components
import Stage from './Stage';
import Display from './Display';
import StartButton from './StartButton';
import { usePlayer } from '../hooks/usePlayer';
import { useStage } from '../hooks/useStage';
import { StyledTetrisWrapper, StyledTetris } from './styles/StyledTetris';

function Tetris() {
  const [dropTime, setDropTime] = useState(null);
  const [gameOver, setGameOver] = useState(false);

  const [player, updatePlayerPos, resetPlayer] = usePlayer();
  const [stage, setStage] = useStage(player);

  console.log('re-render');
  const movePlayer = (dir: number) => {
    updatePlayerPos({ x: dir, y: 0 });
  };
  const startGame = () => {
    /// reset everything
    setStage(createStage());
    resetPlayer();
  };
  
  const drop = () => {
    updatePlayerPos({ x: 0, y: 1, collided: false });
  };
  
  const dropPlayer = () => {
    drop();
  };

  const move = ({ keyCode }: { keyCode: number }) => {
    console.log(keyCode);
    if (!gameOver) {
      if (keyCode === 37) {
        movePlayer(-1);
      } else if (keyCode === 39) {
        movePlayer(1);
      } else if (keyCode === 40) {
        dropPlayer();
      }
    }
  };
  return (
    <StyledTetrisWrapper role="button" tabIndex="0" onKeyDown={(e) => move(e)}>
      <StyledTetris>
        <Stage stage={stage} />
        <aside>
          {gameOver ? (
            <Display gameOver={gameOver} text="Game Over" />
          ) : (
            <div>
              <Display text="Score" gameOver={gameOver} />
              <Display text="Rows" gameOver={gameOver} />
              <Display text="Level1" gameOver={gameOver} />
            </div>
          )}
          <StartButton callBack={() => startGame()} />
        </aside>
      </StyledTetris>
    </StyledTetrisWrapper>
  );
}
export default Tetris;
