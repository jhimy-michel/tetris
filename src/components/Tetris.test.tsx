import { render } from '@testing-library/react';
import Tetris from "./Tetris";

describe("Testing Tetrix main component", () => {
  it("should match snapshot", () => {
    const view = render(<Tetris />);
    expect(view.container).toMatchSnapshot();
  });
});
