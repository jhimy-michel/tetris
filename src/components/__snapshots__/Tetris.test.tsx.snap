// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Tetrix main component should match snapshot 1`] = `
<div>
  <div
    class="sc-eCAqax hUOTtZ"
    role="button"
    tabindex="0"
  >
    <div
      class="sc-jSFipO fJFjjz"
    >
      <div
        class="sc-gtsqUy dAwutc"
        height="20"
        width="12"
      >
        <div
          class="sc-bdnyFh eWIcfs"
          color="0,0,0"
          type="0"
        />
        <div
          class="sc-bdnyFh eWIcfs"
          color="0,0,0"
          type="0"
        />
        <div
          class="sc-bdnyFh eWIcfs"
          color="0,0,0"
          type="0"
        />
        <div
          class="sc-bdnyFh eWIcfs"
          color="0,0,0"
          type="0"
        />
        <div
          class="sc-bdnyFh eWIcfs"
          color="0,0,0"
          type="0"
        />
        <div
          class="sc-bdnyFh eWIcfs"
          color="0,0,0"
          type="0"
        />
        <div
          class="sc-bdnyFh eWIcfs"
          color="0,0,0"
          type="0"
        />
        <div
          class="sc-bdnyFh eWIcfs"
          color="0,0,0"
          type="0"
        />
        <div
          class="sc-bdnyFh eWIcfs"
          color="0,0,0"
          type="0"
        />
        <div
          class="sc-bdnyFh eWIcfs"
          color="0,0,0"
          type="0"
        />
        <div
          class="sc-bdnyFh eWIcfs"
          color="0,0,0"
          type="0"
        />
        <div
          class="sc-bdnyFh eWIcfs"
          color="0,0,0"
          type="0"
        />
        <div
          class="sc-bdnyFh eWIcfs"
          color="0,0,0"
          type="0"
        />
        <div
          class="sc-bdnyFh eWIcfs"
          color="0,0,0"
          type="0"
        />
        <div
          class="sc-bdnyFh eWIcfs"
          color="0,0,0"
          type="0"
        />
        <div
          class="sc-bdnyFh eWIcfs"
          color="0,0,0"
          type="0"
        />
        <div
          class="sc-bdnyFh eWIcfs"
          color="0,0,0"
          type="0"
        />
        <div
          class="sc-bdnyFh eWIcfs"
          color="0,0,0"
          type="0"
        />
        <div
          class="sc-bdnyFh eWIcfs"
          color="0,0,0"
          type="0"
        />
        <div
          class="sc-bdnyFh eWIcfs"
          color="0,0,0"
          type="0"
        />
        <div
          class="sc-bdnyFh eWIcfs"
          color="0,0,0"
          type="0"
        />
        <div
          class="sc-bdnyFh eWIcfs"
          color="0,0,0"
          type="0"
        />
        <div
          class="sc-bdnyFh eWIcfs"
          color="0,0,0"
          type="0"
        />
        <div
          class="sc-bdnyFh eWIcfs"
          color="0,0,0"
          type="0"
        />
        <div
          class="sc-bdnyFh eWIcfs"
          color="0,0,0"
          type="0"
        />
        <div
          class="sc-bdnyFh eWIcfs"
          color="0,0,0"
          type="0"
        />
        <div
          class="sc-bdnyFh eWIcfs"
          color="0,0,0"
          type="0"
        />
        <div
          class="sc-bdnyFh eWIcfs"
          color="0,0,0"
          type="0"
        />
        <div
          class="sc-bdnyFh eWIcfs"
          color="0,0,0"
          type="0"
        />
        <div
          class="sc-bdnyFh eWIcfs"
          color="0,0,0"
          type="0"
        />
        <div
          class="sc-bdnyFh eWIcfs"
          color="0,0,0"
          type="0"
        />
        <div
          class="sc-bdnyFh eWIcfs"
          color="0,0,0"
          type="0"
        />
        <div
          class="sc-bdnyFh eWIcfs"
          color="0,0,0"
          type="0"
        />
        <div
          class="sc-bdnyFh eWIcfs"
          color="0,0,0"
          type="0"
        />
        <div
          class="sc-bdnyFh eWIcfs"
          color="0,0,0"
          type="0"
        />
        <div
          class="sc-bdnyFh eWIcfs"
          color="0,0,0"
          type="0"
        />
        <div
          class="sc-bdnyFh eWIcfs"
          color="0,0,0"
          type="0"
        />
        <div
          class="sc-bdnyFh eWIcfs"
          color="0,0,0"
          type="0"
        />
        <div
          class="sc-bdnyFh eWIcfs"
          color="0,0,0"
          type="0"
        />
        <div
          class="sc-bdnyFh eWIcfs"
          color="0,0,0"
          type="0"
        />
        <div
          class="sc-bdnyFh eWIcfs"
          color="0,0,0"
          type="0"
        />
        <div
          class="sc-bdnyFh eWIcfs"
          color="0,0,0"
          type="0"
        />
        <div
          class="sc-bdnyFh eWIcfs"
          color="0,0,0"
          type="0"
        />
        <div
          class="sc-bdnyFh eWIcfs"
          color="0,0,0"
          type="0"
        />
        <div
          class="sc-bdnyFh eWIcfs"
          color="0,0,0"
          type="0"
        />
        <div
          class="sc-bdnyFh eWIcfs"
          color="0,0,0"
          type="0"
        />
        <div
          class="sc-bdnyFh eWIcfs"
          color="0,0,0"
          type="0"
        />
        <div
          class="sc-bdnyFh eWIcfs"
          color="0,0,0"
          type="0"
        />
        <div
          class="sc-bdnyFh eWIcfs"
          color="0,0,0"
          type="0"
        />
        <div
          class="sc-bdnyFh eWIcfs"
          color="0,0,0"
          type="0"
        />
        <div
          class="sc-bdnyFh eWIcfs"
          color="0,0,0"
          type="0"
        />
        <div
          class="sc-bdnyFh eWIcfs"
          color="0,0,0"
          type="0"
        />
        <div
          class="sc-bdnyFh eWIcfs"
          color="0,0,0"
          type="0"
        />
        <div
          class="sc-bdnyFh eWIcfs"
          color="0,0,0"
          type="0"
        />
        <div
          class="sc-bdnyFh eWIcfs"
          color="0,0,0"
          type="0"
        />
        <div
          class="sc-bdnyFh eWIcfs"
          color="0,0,0"
          type="0"
        />
        <div
          class="sc-bdnyFh eWIcfs"
          color="0,0,0"
          type="0"
        />
        <div
          class="sc-bdnyFh eWIcfs"
          color="0,0,0"
          type="0"
        />
        <div
          class="sc-bdnyFh eWIcfs"
          color="0,0,0"
          type="0"
        />
        <div
          class="sc-bdnyFh eWIcfs"
          color="0,0,0"
          type="0"
        />
        <div
          class="sc-bdnyFh eWIcfs"
          color="0,0,0"
          type="0"
        />
        <div
          class="sc-bdnyFh eWIcfs"
          color="0,0,0"
          type="0"
        />
        <div
          class="sc-bdnyFh eWIcfs"
          color="0,0,0"
          type="0"
        />
        <div
          class="sc-bdnyFh eWIcfs"
          color="0,0,0"
          type="0"
        />
        <div
          class="sc-bdnyFh eWIcfs"
          color="0,0,0"
          type="0"
        />
        <div
          class="sc-bdnyFh eWIcfs"
          color="0,0,0"
          type="0"
        />
        <div
          class="sc-bdnyFh eWIcfs"
          color="0,0,0"
          type="0"
        />
        <div
          class="sc-bdnyFh eWIcfs"
          color="0,0,0"
          type="0"
        />
        <div
          class="sc-bdnyFh eWIcfs"
          color="0,0,0"
          type="0"
        />
        <div
          class="sc-bdnyFh eWIcfs"
          color="0,0,0"
          type="0"
        />
        <div
          class="sc-bdnyFh eWIcfs"
          color="0,0,0"
          type="0"
        />
        <div
          class="sc-bdnyFh eWIcfs"
          color="0,0,0"
          type="0"
        />
        <div
          class="sc-bdnyFh eWIcfs"
          color="0,0,0"
          type="0"
        />
        <div
          class="sc-bdnyFh eWIcfs"
          color="0,0,0"
          type="0"
        />
        <div
          class="sc-bdnyFh eWIcfs"
          color="0,0,0"
          type="0"
        />
        <div
          class="sc-bdnyFh eWIcfs"
          color="0,0,0"
          type="0"
        />
        <div
          class="sc-bdnyFh eWIcfs"
          color="0,0,0"
          type="0"
        />
        <div
          class="sc-bdnyFh eWIcfs"
          color="0,0,0"
          type="0"
        />
        <div
          class="sc-bdnyFh eWIcfs"
          color="0,0,0"
          type="0"
        />
        <div
          class="sc-bdnyFh eWIcfs"
          color="0,0,0"
          type="0"
        />
        <div
          class="sc-bdnyFh eWIcfs"
          color="0,0,0"
          type="0"
        />
        <div
          class="sc-bdnyFh eWIcfs"
          color="0,0,0"
          type="0"
        />
        <div
          class="sc-bdnyFh eWIcfs"
          color="0,0,0"
          type="0"
        />
        <div
          class="sc-bdnyFh eWIcfs"
          color="0,0,0"
          type="0"
        />
        <div
          class="sc-bdnyFh eWIcfs"
          color="0,0,0"
          type="0"
        />
        <div
          class="sc-bdnyFh eWIcfs"
          color="0,0,0"
          type="0"
        />
        <div
          class="sc-bdnyFh eWIcfs"
          color="0,0,0"
          type="0"
        />
        <div
          class="sc-bdnyFh eWIcfs"
          color="0,0,0"
          type="0"
        />
        <div
          class="sc-bdnyFh eWIcfs"
          color="0,0,0"
          type="0"
        />
        <div
          class="sc-bdnyFh eWIcfs"
          color="0,0,0"
          type="0"
        />
        <div
          class="sc-bdnyFh eWIcfs"
          color="0,0,0"
          type="0"
        />
        <div
          class="sc-bdnyFh eWIcfs"
          color="0,0,0"
          type="0"
        />
        <div
          class="sc-bdnyFh eWIcfs"
          color="0,0,0"
          type="0"
        />
        <div
          class="sc-bdnyFh eWIcfs"
          color="0,0,0"
          type="0"
        />
        <div
          class="sc-bdnyFh eWIcfs"
          color="0,0,0"
          type="0"
        />
        <div
          class="sc-bdnyFh eWIcfs"
          color="0,0,0"
          type="0"
        />
        <div
          class="sc-bdnyFh eWIcfs"
          color="0,0,0"
          type="0"
        />
        <div
          class="sc-bdnyFh eWIcfs"
          color="0,0,0"
          type="0"
        />
        <div
          class="sc-bdnyFh eWIcfs"
          color="0,0,0"
          type="0"
        />
        <div
          class="sc-bdnyFh eWIcfs"
          color="0,0,0"
          type="0"
        />
        <div
          class="sc-bdnyFh eWIcfs"
          color="0,0,0"
          type="0"
        />
        <div
          class="sc-bdnyFh eWIcfs"
          color="0,0,0"
          type="0"
        />
        <div
          class="sc-bdnyFh eWIcfs"
          color="0,0,0"
          type="0"
        />
        <div
          class="sc-bdnyFh eWIcfs"
          color="0,0,0"
          type="0"
        />
        <div
          class="sc-bdnyFh eWIcfs"
          color="0,0,0"
          type="0"
        />
        <div
          class="sc-bdnyFh eWIcfs"
          color="0,0,0"
          type="0"
        />
        <div
          class="sc-bdnyFh eWIcfs"
          color="0,0,0"
          type="0"
        />
        <div
          class="sc-bdnyFh eWIcfs"
          color="0,0,0"
          type="0"
        />
        <div
          class="sc-bdnyFh eWIcfs"
          color="0,0,0"
          type="0"
        />
        <div
          class="sc-bdnyFh eWIcfs"
          color="0,0,0"
          type="0"
        />
        <div
          class="sc-bdnyFh eWIcfs"
          color="0,0,0"
          type="0"
        />
        <div
          class="sc-bdnyFh eWIcfs"
          color="0,0,0"
          type="0"
        />
        <div
          class="sc-bdnyFh eWIcfs"
          color="0,0,0"
          type="0"
        />
        <div
          class="sc-bdnyFh eWIcfs"
          color="0,0,0"
          type="0"
        />
        <div
          class="sc-bdnyFh eWIcfs"
          color="0,0,0"
          type="0"
        />
        <div
          class="sc-bdnyFh eWIcfs"
          color="0,0,0"
          type="0"
        />
        <div
          class="sc-bdnyFh eWIcfs"
          color="0,0,0"
          type="0"
        />
        <div
          class="sc-bdnyFh eWIcfs"
          color="0,0,0"
          type="0"
        />
        <div
          class="sc-bdnyFh eWIcfs"
          color="0,0,0"
          type="0"
        />
        <div
          class="sc-bdnyFh eWIcfs"
          color="0,0,0"
          type="0"
        />
        <div
          class="sc-bdnyFh eWIcfs"
          color="0,0,0"
          type="0"
        />
        <div
          class="sc-bdnyFh eWIcfs"
          color="0,0,0"
          type="0"
        />
        <div
          class="sc-bdnyFh eWIcfs"
          color="0,0,0"
          type="0"
        />
        <div
          class="sc-bdnyFh eWIcfs"
          color="0,0,0"
          type="0"
        />
        <div
          class="sc-bdnyFh eWIcfs"
          color="0,0,0"
          type="0"
        />
        <div
          class="sc-bdnyFh eWIcfs"
          color="0,0,0"
          type="0"
        />
        <div
          class="sc-bdnyFh eWIcfs"
          color="0,0,0"
          type="0"
        />
        <div
          class="sc-bdnyFh eWIcfs"
          color="0,0,0"
          type="0"
        />
        <div
          class="sc-bdnyFh eWIcfs"
          color="0,0,0"
          type="0"
        />
        <div
          class="sc-bdnyFh eWIcfs"
          color="0,0,0"
          type="0"
        />
        <div
          class="sc-bdnyFh eWIcfs"
          color="0,0,0"
          type="0"
        />
        <div
          class="sc-bdnyFh eWIcfs"
          color="0,0,0"
          type="0"
        />
        <div
          class="sc-bdnyFh eWIcfs"
          color="0,0,0"
          type="0"
        />
        <div
          class="sc-bdnyFh eWIcfs"
          color="0,0,0"
          type="0"
        />
        <div
          class="sc-bdnyFh eWIcfs"
          color="0,0,0"
          type="0"
        />
        <div
          class="sc-bdnyFh eWIcfs"
          color="0,0,0"
          type="0"
        />
        <div
          class="sc-bdnyFh eWIcfs"
          color="0,0,0"
          type="0"
        />
        <div
          class="sc-bdnyFh eWIcfs"
          color="0,0,0"
          type="0"
        />
        <div
          class="sc-bdnyFh eWIcfs"
          color="0,0,0"
          type="0"
        />
        <div
          class="sc-bdnyFh eWIcfs"
          color="0,0,0"
          type="0"
        />
        <div
          class="sc-bdnyFh eWIcfs"
          color="0,0,0"
          type="0"
        />
        <div
          class="sc-bdnyFh eWIcfs"
          color="0,0,0"
          type="0"
        />
        <div
          class="sc-bdnyFh eWIcfs"
          color="0,0,0"
          type="0"
        />
        <div
          class="sc-bdnyFh eWIcfs"
          color="0,0,0"
          type="0"
        />
        <div
          class="sc-bdnyFh eWIcfs"
          color="0,0,0"
          type="0"
        />
        <div
          class="sc-bdnyFh eWIcfs"
          color="0,0,0"
          type="0"
        />
        <div
          class="sc-bdnyFh eWIcfs"
          color="0,0,0"
          type="0"
        />
        <div
          class="sc-bdnyFh eWIcfs"
          color="0,0,0"
          type="0"
        />
        <div
          class="sc-bdnyFh eWIcfs"
          color="0,0,0"
          type="0"
        />
        <div
          class="sc-bdnyFh eWIcfs"
          color="0,0,0"
          type="0"
        />
        <div
          class="sc-bdnyFh eWIcfs"
          color="0,0,0"
          type="0"
        />
        <div
          class="sc-bdnyFh eWIcfs"
          color="0,0,0"
          type="0"
        />
        <div
          class="sc-bdnyFh eWIcfs"
          color="0,0,0"
          type="0"
        />
        <div
          class="sc-bdnyFh eWIcfs"
          color="0,0,0"
          type="0"
        />
        <div
          class="sc-bdnyFh eWIcfs"
          color="0,0,0"
          type="0"
        />
        <div
          class="sc-bdnyFh eWIcfs"
          color="0,0,0"
          type="0"
        />
        <div
          class="sc-bdnyFh eWIcfs"
          color="0,0,0"
          type="0"
        />
        <div
          class="sc-bdnyFh eWIcfs"
          color="0,0,0"
          type="0"
        />
        <div
          class="sc-bdnyFh eWIcfs"
          color="0,0,0"
          type="0"
        />
        <div
          class="sc-bdnyFh eWIcfs"
          color="0,0,0"
          type="0"
        />
        <div
          class="sc-bdnyFh eWIcfs"
          color="0,0,0"
          type="0"
        />
        <div
          class="sc-bdnyFh eWIcfs"
          color="0,0,0"
          type="0"
        />
        <div
          class="sc-bdnyFh eWIcfs"
          color="0,0,0"
          type="0"
        />
        <div
          class="sc-bdnyFh eWIcfs"
          color="0,0,0"
          type="0"
        />
        <div
          class="sc-bdnyFh eWIcfs"
          color="0,0,0"
          type="0"
        />
        <div
          class="sc-bdnyFh eWIcfs"
          color="0,0,0"
          type="0"
        />
        <div
          class="sc-bdnyFh eWIcfs"
          color="0,0,0"
          type="0"
        />
        <div
          class="sc-bdnyFh eWIcfs"
          color="0,0,0"
          type="0"
        />
        <div
          class="sc-bdnyFh eWIcfs"
          color="0,0,0"
          type="0"
        />
        <div
          class="sc-bdnyFh eWIcfs"
          color="0,0,0"
          type="0"
        />
        <div
          class="sc-bdnyFh eWIcfs"
          color="0,0,0"
          type="0"
        />
        <div
          class="sc-bdnyFh eWIcfs"
          color="0,0,0"
          type="0"
        />
        <div
          class="sc-bdnyFh eWIcfs"
          color="0,0,0"
          type="0"
        />
        <div
          class="sc-bdnyFh eWIcfs"
          color="0,0,0"
          type="0"
        />
        <div
          class="sc-bdnyFh eWIcfs"
          color="0,0,0"
          type="0"
        />
        <div
          class="sc-bdnyFh eWIcfs"
          color="0,0,0"
          type="0"
        />
        <div
          class="sc-bdnyFh eWIcfs"
          color="0,0,0"
          type="0"
        />
        <div
          class="sc-bdnyFh eWIcfs"
          color="0,0,0"
          type="0"
        />
        <div
          class="sc-bdnyFh eWIcfs"
          color="0,0,0"
          type="0"
        />
        <div
          class="sc-bdnyFh eWIcfs"
          color="0,0,0"
          type="0"
        />
        <div
          class="sc-bdnyFh eWIcfs"
          color="0,0,0"
          type="0"
        />
        <div
          class="sc-bdnyFh eWIcfs"
          color="0,0,0"
          type="0"
        />
        <div
          class="sc-bdnyFh eWIcfs"
          color="0,0,0"
          type="0"
        />
        <div
          class="sc-bdnyFh eWIcfs"
          color="0,0,0"
          type="0"
        />
        <div
          class="sc-bdnyFh eWIcfs"
          color="0,0,0"
          type="0"
        />
        <div
          class="sc-bdnyFh eWIcfs"
          color="0,0,0"
          type="0"
        />
        <div
          class="sc-bdnyFh eWIcfs"
          color="0,0,0"
          type="0"
        />
        <div
          class="sc-bdnyFh eWIcfs"
          color="0,0,0"
          type="0"
        />
        <div
          class="sc-bdnyFh eWIcfs"
          color="0,0,0"
          type="0"
        />
        <div
          class="sc-bdnyFh eWIcfs"
          color="0,0,0"
          type="0"
        />
        <div
          class="sc-bdnyFh eWIcfs"
          color="0,0,0"
          type="0"
        />
        <div
          class="sc-bdnyFh eWIcfs"
          color="0,0,0"
          type="0"
        />
        <div
          class="sc-bdnyFh eWIcfs"
          color="0,0,0"
          type="0"
        />
        <div
          class="sc-bdnyFh eWIcfs"
          color="0,0,0"
          type="0"
        />
        <div
          class="sc-bdnyFh eWIcfs"
          color="0,0,0"
          type="0"
        />
        <div
          class="sc-bdnyFh eWIcfs"
          color="0,0,0"
          type="0"
        />
        <div
          class="sc-bdnyFh eWIcfs"
          color="0,0,0"
          type="0"
        />
        <div
          class="sc-bdnyFh eWIcfs"
          color="0,0,0"
          type="0"
        />
        <div
          class="sc-bdnyFh eWIcfs"
          color="0,0,0"
          type="0"
        />
        <div
          class="sc-bdnyFh eWIcfs"
          color="0,0,0"
          type="0"
        />
        <div
          class="sc-bdnyFh eWIcfs"
          color="0,0,0"
          type="0"
        />
        <div
          class="sc-bdnyFh eWIcfs"
          color="0,0,0"
          type="0"
        />
        <div
          class="sc-bdnyFh eWIcfs"
          color="0,0,0"
          type="0"
        />
        <div
          class="sc-bdnyFh eWIcfs"
          color="0,0,0"
          type="0"
        />
        <div
          class="sc-bdnyFh eWIcfs"
          color="0,0,0"
          type="0"
        />
        <div
          class="sc-bdnyFh eWIcfs"
          color="0,0,0"
          type="0"
        />
        <div
          class="sc-bdnyFh eWIcfs"
          color="0,0,0"
          type="0"
        />
        <div
          class="sc-bdnyFh eWIcfs"
          color="0,0,0"
          type="0"
        />
        <div
          class="sc-bdnyFh eWIcfs"
          color="0,0,0"
          type="0"
        />
        <div
          class="sc-bdnyFh eWIcfs"
          color="0,0,0"
          type="0"
        />
        <div
          class="sc-bdnyFh eWIcfs"
          color="0,0,0"
          type="0"
        />
        <div
          class="sc-bdnyFh eWIcfs"
          color="0,0,0"
          type="0"
        />
        <div
          class="sc-bdnyFh eWIcfs"
          color="0,0,0"
          type="0"
        />
        <div
          class="sc-bdnyFh eWIcfs"
          color="0,0,0"
          type="0"
        />
        <div
          class="sc-bdnyFh eWIcfs"
          color="0,0,0"
          type="0"
        />
        <div
          class="sc-bdnyFh eWIcfs"
          color="0,0,0"
          type="0"
        />
        <div
          class="sc-bdnyFh eWIcfs"
          color="0,0,0"
          type="0"
        />
        <div
          class="sc-bdnyFh eWIcfs"
          color="0,0,0"
          type="0"
        />
        <div
          class="sc-bdnyFh eWIcfs"
          color="0,0,0"
          type="0"
        />
        <div
          class="sc-bdnyFh eWIcfs"
          color="0,0,0"
          type="0"
        />
        <div
          class="sc-bdnyFh eWIcfs"
          color="0,0,0"
          type="0"
        />
        <div
          class="sc-bdnyFh eWIcfs"
          color="0,0,0"
          type="0"
        />
        <div
          class="sc-bdnyFh eWIcfs"
          color="0,0,0"
          type="0"
        />
        <div
          class="sc-bdnyFh eWIcfs"
          color="0,0,0"
          type="0"
        />
        <div
          class="sc-bdnyFh eWIcfs"
          color="0,0,0"
          type="0"
        />
        <div
          class="sc-bdnyFh eWIcfs"
          color="0,0,0"
          type="0"
        />
        <div
          class="sc-bdnyFh eWIcfs"
          color="0,0,0"
          type="0"
        />
        <div
          class="sc-bdnyFh eWIcfs"
          color="0,0,0"
          type="0"
        />
        <div
          class="sc-bdnyFh eWIcfs"
          color="0,0,0"
          type="0"
        />
        <div
          class="sc-bdnyFh eWIcfs"
          color="0,0,0"
          type="0"
        />
        <div
          class="sc-bdnyFh eWIcfs"
          color="0,0,0"
          type="0"
        />
        <div
          class="sc-bdnyFh eWIcfs"
          color="0,0,0"
          type="0"
        />
        <div
          class="sc-bdnyFh eWIcfs"
          color="0,0,0"
          type="0"
        />
        <div
          class="sc-bdnyFh eWIcfs"
          color="0,0,0"
          type="0"
        />
        <div
          class="sc-bdnyFh eWIcfs"
          color="0,0,0"
          type="0"
        />
        <div
          class="sc-bdnyFh eWIcfs"
          color="0,0,0"
          type="0"
        />
        <div
          class="sc-bdnyFh eWIcfs"
          color="0,0,0"
          type="0"
        />
        <div
          class="sc-bdnyFh eWIcfs"
          color="0,0,0"
          type="0"
        />
        <div
          class="sc-bdnyFh eWIcfs"
          color="0,0,0"
          type="0"
        />
        <div
          class="sc-bdnyFh eWIcfs"
          color="0,0,0"
          type="0"
        />
      </div>
      <aside>
        <div>
          <div
            class="sc-dlniIP hTTtfJ"
          >
            Score: 0
          </div>
          <div
            class="sc-dlniIP hTTtfJ"
          >
            Rows: 0
          </div>
          <div
            class="sc-dlniIP hTTtfJ"
          >
            Level: 0
          </div>
        </div>
        <div
          class="sc-hKFymg pDXxZ"
        >
          Start Game
        </div>
      </aside>
    </div>
  </div>
</div>
`;
